File: software/examples/c/example_configuration.c
#include <stdio.h>

#include "ip_connection.h"
#include "brick_dc.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XXYYZZ" // Change XXYYZZ to the UID of your DC Brick

int main(void) {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	DC dc;
	dc_create(&dc, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		return 1;
	}
	// Don't use device before ipcon is connected

	dc_set_drive_mode(&dc, DC_DRIVE_MODE_DRIVE_COAST);
	dc_set_pwm_frequency(&dc, 10000); // Use PWM frequency of 10kHz
	dc_set_acceleration(&dc, 5000); // Slow acceleration
	dc_set_velocity(&dc, 32767); // Full speed forward
	dc_enable(&dc); // Enable motor power

	printf("Press key to exit\n");
	getchar();
	dc_disable(&dc); // Disable motor power
	dc_destroy(&dc);
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
	return 0;
}

File: software/examples/go/example_configuration.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/dc_brick"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"time"
)

const ADDR string = "localhost:4223"
const UID string = "XXYYZZ" // Change XXYYZZ to the UID of your DC Brick.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	dc, _ := dc_brick.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	dc.SetDriveMode(dc_brick.DriveModeDriveCoast)
	dc.SetPWMFrequency(10000) // Use PWM frequency of 10 kHz
	dc.SetAcceleration(4096)  // Slow acceleration (12.5 %/s)
	dc.SetVelocity(32767)     // Full speed forward (100 %)
	dc.Enable()               // Enable motor power

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

	// Stop motor before disabling motor power
	dc.SetAcceleration(16384)           // Fast decceleration (50 %/s) for stopping
	dc.SetVelocity(0)                   // Request motor stop
	time.Sleep(2000 * time.Millisecond) // Wait for motor to actually stop: velocity (100 %) / decceleration (50 %/s) = 2 s
	dc.Disable()                        // Disable motor power
}
